pandas/io/excel.py

```diff
@@ -1610,35 +1672,57 @@ def _convert_to_style(self, style_dict, num_format_str=None):
             return None
 
         # Create a XlsxWriter format object.
-        xl_format = self.book.add_format()
+        props = {}
 
         if num_format_str is not None:
-            xl_format.set_num_format(num_format_str)
+            props['num_format'] = num_format_str
 
```

Comment:
yes i think think excel should be refactored into a subdir of writer code and style things should live there

maybe make an issue about this
it's a bit of work to split it then adding things like style should be easy